pinot-core/src/main/java/org/apache/pinot/core/plan/PlanNode.java

```diff
@@ -18,16 +18,16 @@
  */
 package org.apache.pinot.core.plan;
 
-import org.apache.pinot.core.common.Block;
 import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.query.spi.data.Block;
 import org.apache.pinot.spi.annotations.InterfaceAudience;
 
 
 /**
  * The <code>PlanNode</code> is a single execution plan node inside the {@link Plan} tree.
  */
 @InterfaceAudience.Private
-public interface PlanNode {
+public interface PlanNode extends org.apache.pinot.query.spi.plan.PlanNode {
```

Comment:
TODO item1:
This cannot be cleanly factor out because of Operator issue. Ideally this should be
```
public interface PlanNode<O extends Operator> {
  O<? extends Block> run();
}